server/src/main/java/org/elasticsearch/transport/OutboundHandler.java

```diff
@@ -194,10 +195,17 @@ public void onResponse(Void v) {
 
                 @Override
                 public void onFailure(Exception e) {
-                    if (NetworkExceptionHelper.isCloseConnectionException(e)) {
-                        logger.debug(() -> new ParameterizedMessage("send message failed [channel: {}]", channel), e);
+                    final Level closeConnectionExceptionLevel = NetworkExceptionHelper.getCloseConnectionExceptionLevel(e);
+                    if (closeConnectionExceptionLevel == Level.OFF) {
+                        logger.warn(new ParameterizedMessage("send message failed [channel: {}]", channel), e);
```

Comment:
Should the condition above be `closeConnectionExceptionLevel == Level.WARN`?

Reply:
No, the `WARN` case is handled in the final `else` branch. This branch is handling the case that the exception doesn't appear to be network-related, and preserves the earlier behaviour that such exceptions are logged as warnings.

Reply:
Oh, I was confused with `Level.OFF` actually representing a `non network related`.
I wonder if that could be changed to `Level.WARN` then, the first branch could be merged with the last one, unless I miss something.

Reply:
Kind of, but I think that risks future bugs. The intention is to distinguish "this exception is network-related and deserves a warning" from "this exception is not network-related". An earlier iteration actually returned `WARN` sometimes but then I moved them all to `INFO` because I found that all the Linux ones might be benign and I didn't want to emit warnings for benign problems. I didn't investigate the Windows behaviour and we might decide to move them back up to `WARN` in future.

I could have used an `Optional<Level>` or a `null` or similar but `Level#OFF` seemed like a simpler way to say the same thing.